### PR TITLE
TELCODOCS-1426 - add HTTP transport to openshift-4.12

### PR DESCRIPTION
**Pull request by aireilly**
Cherry-picks HTTP transport docs + PV/PVC > ConfigMap PTP events docs to openshift-4.12 branch.

Version(s):
openshift-4.12

To summarize: 

HTTP transport and using ConfigMap events feature has been backported to 4.12 from 4.13.

* The primary PR that got merged to main/CPed to 4.13 is [this one](https://github.com/openshift/openshift-docs/pull/55695).
*  in 4.13, a subpotimal implementation of events storage was implemented using PV/PVC CRs. This work has been changed so that manual PV/PVC creation is required. No change to the feature for the customer side - they don't need to do anything. [Orig PR is here.](https://github.com/openshift/openshift-docs/pull/62938)
* On this PR, there are a number of small commits to adjust version numbers.  

Issue:
https://issues.redhat.com/browse/TELCODOCS-1426

Link to docs preview:
* https://63051--docspreview.netlify.app/openshift-enterprise/latest/networking/using-ptp#ptp-hardware-fast-event-notifications-framework
* https://63051--docspreview.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp_far_edge/ztp-advanced-policy-config#ztp-advanced-policy-config-ptp_ztp-advanced-policy-config
* https://63051--docspreview.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp_far_edge/ztp-advanced-policy-config#ztp-advanced-policy-config-bare-metal_ztp-advanced-policy-config
* https://63051--docspreview.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-12-release-notes#ocp-4-12-security
* https://63051--docspreview.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-12-release-notes#ocp-4-12-ran-http-transport

QE review:
- [x] QE has approved this change.

### CODE REVIEW

**Review by aneeshkp — APPROVED** [review]
/lgtm

File: modules/cnf-configuring-the-ptp-fast-event-publisher.adoc [hunk 1]
```diff
@@ -32,10 +34,14 @@ spec:
...
   ptpEventConfig:
     enableEventPublisher: true <1>
-    transportHost: amqp://<instance_name>.<namespace>.svc.cluster.local <2>
 ----
 <1> Set `enableEventPublisher` to `true` to enable PTP fast event notifications.
-<2> Set `transportHost` to the AMQ router that you configured where `<instance_name>` and `<namespace>` correspond to the AMQ Interconnect router instance name and namespace, for example, `amqp://amq-interconnect.amq-interconnect.svc.cluster.local`
++
+[NOTE]
+====
+In {product-title} 4.13 or later, you do not need to set the `spec.ptpEventConfig.transportHost` field in the `PtpOperatorConfig` resource when you use HTTP transport for PTP events.
```
Comment (jzding) [inline modules/cnf-configuring-the-ptp-fast-event-publisher.adoc:42]: this should be `4.12 or later` now.

File: modules/nw-rfhe-creating-hardware-event.adoc [hunk 2]
```diff
@@ -35,16 +38,23 @@ metadata:
...
++
+--
 <1> Required. Use the `nodeSelector` field to target nodes with the specified label, for example, `node-role.kubernetes.io/hw-event: ""`.
-<2> Required. AMQP host that delivers the events at the transport layer using the AMQP protocol.
-<3> Optional. The default value is `debug`. Sets the log level in `hw-event-proxy` logs. The following log levels are available: `fatal`, `error`, `warning`, `info`, `debug`, `trace`.
-<4> Optional. Sets the timeout value in milliseconds for the Message Parser. If a message parsing request is not responded to within the timeout duration, the original hardware event message is passed to the cloud native event framework. The default value is 10.
++
+[NOTE]
+====
+In {product-title} 4.13 or later, you do not need to set the `spec.transportHost` field in the `HardwareEvent` resource when you use HTTP transport for bare-metal events.
```
Comment (jzding) [inline modules/nw-rfhe-creating-hardware-event.adoc:50]: this should be `4.12 or later` now.

File: modules/ztp-configuring-hwevents-using-pgt.adoc [hunk 3]
```diff
@@ -36,34 +31,27 @@ You can configure bare-metal hardware events for vRAN clusters that are deployed
...
     logLevel: "info"
 ----
-<1>  The `transportHost` URL is composed of the existing AMQ Interconnect CR `name` and `namespace`. For example, in `transportHost: "amqp://amq-router.amq-router.svc.cluster.local"`, the AMQ Interconnect `name` and `namespace` are both set to `amq-router`.
-
+<1> Each baseboard management controller (BMC) requires a single `HardwareEvent` CR only.
 +
 [NOTE]
 ====
-Each baseboard management controller (BMC) requires a single `HardwareEvent` resource only.
+In {product-title} 4.13 or later, you do not need to set the `transportHost` field in the `HardwareEvent` custom resource (CR) when you use HTTP transport with bare-metal events.
```
Comment (jzding) [inline modules/ztp-configuring-hwevents-using-pgt.adoc:49]: this should be `4.12 or later` now.

File: modules/ztp-configuring-ptp-fast-events.adoc [hunk 4]
```diff
@@ -2,40 +2,41 @@
...
+  spec:
+    daemonNodeSelector: {}
+    ptpEventConfig:
+      enableEventPublisher: true
+      transportHost: http://ptp-event-publisher-service-NODE_NAME.openshift-ptp.svc.cluster.local:9043
 ----
++
+[NOTE]
+====
+In {product-title} 4.13 or later, you do not need to set the `transportHost` field in the `PtpOperatorConfig` resource when you use HTTP transport with PTP events.
```
Comment (jzding) [inline modules/ztp-configuring-ptp-fast-events.adoc:38]: this should be `4.12 or later` now.

**Review by jzding — APPROVED** [review]
/lgtm

**Comment by abhatt-rh** [comment]
> /label merge-review-needed

Hi @aireilly, I don't see `peer-review` related labels. Wondering if you wanted to request a peer-review instead of merge-review?

Removing the `merge-review-in-progress` label for now.

**Comment by aireilly** [comment]
It's a CP to 4.12 from main. The content has been peer reviewed on main @abhatt-rh.

**Comment by aireilly** [comment]
I guess there are some very minor changes.

**Comment by abhatt-rh** [comment]
> It's a CP to 4.12 from main. The content has been peer reviewed on main @abhatt-rh.

Sorry, I didn't see a commit ID from where it was originally CP'ed, which made me think that this might be a candidate for peer review. I see that you have added `peer-review-needed` label, yet it might help if you could add a link or refer to the original PR/commit. Thanks!

**Comment by aireilly** [comment]
@abhatt-rh See updated first post.

**Comment by tradej** [comment]
LGTM from Docs (change management)

File: modules/cnf-about-ptp-fast-event-notifications-framework.adoc [hunk 5]
```diff
@@ -6,24 +6,42 @@
...
+The `cloud-event-proxy` sidecar container can access the same resources as the primary application container without using any of the resources of the primary application and with no significant latency.
 
-The `cloud-event-proxy` sidecar container can access the same resources as the primary vRAN application without using any of the resources of the primary application and with no significant latency.
+include::snippets/ptp-amq-interconnect-eol.adoc[]
 
-The fast events notifications framework uses a REST API for communication and is based on the O-RAN REST API specification. The framework consists of a publisher, subscriber, and an AMQ messaging bus to handle communications between the publisher and subscriber applications. The `cloud-event-proxy` sidecar is a utility container that runs in a pod that is loosely coupled to the main DU application container on the DU node. It provides an event publishing framework that allows you to subscribe DU applications to published PTP events.
+.Overview of PTP fast events
+image::218_OpenShift_PTP_events_0222.png[Overview of PTP fast events]
```
Comment (abhatt-rh) [inline modules/cnf-about-ptp-fast-event-notifications-framework.adoc:23]: @aireilly Probably this one needs to be updated to 319?

**Review by abhatt-rh — COMMENTED** [review]
Hi @aireilly I have completed reviewing and comparing the changes for this PR with the [4.13 CP'ed version](https://github.com/openshift/openshift-docs/pull/55695) and its [follow-up updated version](https://github.com/openshift/openshift-docs/pull/62938). Since these are already reviewed and merged to main, I did not have any comment aside from the one I already added.

Great work backporting these to 4.12; some really tricky bits there!

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

**Comment by aireilly** [comment]
Waiting on change management acks before merging.

**Comment by aireilly** [comment]
/label merge-review-needed

All acks are in.

**Pull request merged**